engine/chainstate/broadcast_utils.go

```diff
@@ -1,76 +1,58 @@
 package chainstate
 
 import (
-	"fmt"
+	"errors"
 	"strings"
-	"sync"
 )
 
-// struct handles communication with the client - returns first successful broadcast
-type broadcastStatus struct {
-	mu          *sync.Mutex
-	complete    bool
-	success     bool
-	syncChannel chan string
-}
-
-func newBroadcastStatus(synchChannel chan string) *broadcastStatus {
-	return &broadcastStatus{complete: false, syncChannel: synchChannel, mu: &sync.Mutex{}}
-}
-
-func (g *broadcastStatus) tryCompleteWithSuccess(fastestProvider string) {
-	g.mu.Lock()
-	defer g.mu.Unlock()
-
-	if !g.complete {
-		g.complete = true
-		g.success = true
-
-		g.syncChannel <- fastestProvider
-		close(g.syncChannel)
+// doesErrorContain will look at a string for a list of strings
+func doesErrorContain(err string, messages []string) bool {
+	lower := strings.ToLower(err)
+	for _, str := range messages {
+		if strings.Contains(lower, str) {
+			return true
+		}
 	}
-
-	// g.mu.Unlock() is done by defer
+	return false
 }
 
-func (g *broadcastStatus) dispose() {
-	g.mu.Lock()
-	defer g.mu.Unlock()
+func groupBroadcastResults(results []*BroadcastResult) *BroadcastResult {
+	var grouped *BroadcastResult
 
-	if !g.complete {
-		g.complete = true
-		close(g.syncChannel) // have to close the channel here to not block client
+	if len(results) == 1 {
+		grouped = results[0]
+	} else {
+		grouped = &BroadcastResult{
+			Provider: ProviderAll,
+			Failure:  groupBroadcastFailures(results),
+		}
 	}
 
-	// g.mu.Unlock() is done by defer
+	return grouped
 }
```

Comment:
Consider:
```golang
func groupBroadcastResults(results []*BroadcastResult) *BroadcastResult {
	if len(results) == 1 {
		return results[0]
	} 
	return &BroadcastResult{
		Provider: ProviderAll,
		Failure:  groupBroadcastFailures(results),
	}
}
```

Just a suggestion :D

Reply:
It  should return nil, as it currently does

Reply:
changed